File: back-end/ImageLinkerHandler.py
class ImageLinkerHandler():
    def __init__(self, data_access):
        self.data_access = data_access

    def delete_link(self, link):
         self.data_access.delete_link(link)

    def upsert_link(self, link):
        self.data_access.upsert_link(link)

    def get_links_for_image(self, image):
        return self.data_access.get_links_for_image(image)

File: front-end/src/JS/ImageLinkerHandler.js
import * as proxy from "./proxy";
import ImageLinker from "../JS/ImageLinker";
class ImageLinkerHandler {
  upsertLink(link) {
    proxy.upsertLink(link);
  }

  getAllLinksToImage(origin) {
    return proxy.getAllLinksToImage(origin);
  }

  deleteLink(link) {
    return proxy.deleteLink(link);
  }

  getConnections(image) {
    const linkHandler = new ImageLinkerHandler();
    return linkHandler.getAllLinksToImage(image).then((connections) => {
      const connectionsToLinks = (connections) =>
        connections.map(
          (connection) =>
            new ImageLinker(
              connection.origin,
              connection.destination,
              connection.destinationImagePath,
              connection.latitude,
              connection.longitude
            )
        );
      return connections ? connectionsToLinks(connections) : [];
    });
  }
}

export default ImageLinkerHandler;
